### PR TITLE
Task4610 report

### PR DESCRIPTION
**Pull request by chuxuanthang**
https://edu-redmine.framgia.vn/issues/4610

### CODE REVIEW

**Comment by chuxuanthang** [comment]
ready

File: database/migrations/2019_01_06_042338_create_restructure_reports_table.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropForeign('users_post_id_foreign');
```
Comment (BunnyPi04) [inline database/migrations/2019_01_06_042338_create_restructure_reports_table.php]: chưa đủ

**Comment by chuxuanthang** [comment]
ready

**Comment by chuxuanthang** [comment]
ready<?php

namespace App;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Answer extends Model
{
    use SoftDeletes;
    
    protected $dates = ['deleted_at'];
    
    protected $fillable = [
        'post_id',
        'user_id',
        'content',
        'total_vote',
        'status',
    ];

    public function user()
    {
        return $this->belongsTo('App\User');
    }

    public function post()
    {
        return $this->belongsTo('App\Post');
    }

    public function replies()
    {
        return $this->morphMany('App\Reply', 'repliable');
    }

    public function votes()
    {
        return $this->morphMany('App\Vote', 'voteable');
    }
    
    public static function boot()
    {
        parent::boot();
        static::deleting(function ($answer) {
            $answer->replies()->delete();
            $answer->votes()->delete();
        });
    }
}

**Comment by chuxuanthang** [comment]
ready

**Pull request merged**